docs/source/design/clis.rst

```diff
@@ -22,14 +22,54 @@ The client code is located in ``flytekit/clients`` and there are two.
 Command Line Interfaces
 ***********************
 
-Flyte CLI
+Flytectl
 ===========
-``flyte-cli`` is the general CLI that can be used to talk to the Flyte control plane (Flyte Admin). It ships with Flytekit as part of the Pypi package. Think of this as the ``kubectl`` for Flyte. In fact, we're working on ``flytectl`` which is under active development the completion of which will deprecate this CLI.
-
-Think of this CLI as a network-aware (i.e. can talk to Admin) but not code-aware (doesn't need to have user code checked out) CLI. In the registration flow, this CLI is responsible for shipping the compiled Protobuf files off to Flyte Admin.
+``flytectl`` is the general CLI that can be used to communicate with the Flyte control plane (FlyteAdmin). It ships with flytekit as part of the Pypi package. Think of this as the ``kubectl`` for Flyte. 
+Think of this CLI as a network-aware (can talk to Admin) but not code-aware (doesn't need to have user code checked out) CLI. In the registration flow, this CLI is responsible for shipping the compiled Protobuf files off to Flyte Admin.
 
 Pyflyte
 =========
-Unlike Flyte CLI, think of this CLI as code-aware, but not network-aware (the latter is not entirely true, but it's helpful to think of it that way).
+Unlike flytectl, think of this CLI as code-aware, but not network-aware (the latter is not entirely true, but it's helpful to think of it that way).
+
+This CLI is responsible for the serialization (compilation) step in the registration flow. It will parse through the user code, looking for tasks, workflows and launch plans, and compile them to protobuf files.
+
+What is ``pyflyte register``?
+=============================
+``pyflyte register`` is a command that registers all the workflows present in the repository/directory using fast-registration. It is equivalent to using two commands (``pyflyte package`` and ``flytectl register``) to perform the same operation (registration). 
+The ``pyflyte package`` command parses and compiles the user’s Python code into Flyte protobuf objects.
+The ``flytectl register`` command ships the protobuf objects over the network to Flyte control plane, wherein ``flytectl`` also allows adding attributes such as IAM role, K8s service account, and so on. 
+
+Why should you use ``pyflyte register``?
```

Comment:
also add that pyflyte register is only fast register for now.  that is, it will always bundle up and upload the code, there is no way around that.  We're going to add a non-fast version shortly.

Reply:
I have mentioned it at the beginning. In the "What is ``pyflyte register`` section?" @wild-endeavor